core/src/main/java/org/hibernate/ogm/model/spi/Association.java

```diff
@@ -184,7 +184,7 @@ public int size() {
 		}
 		else {
 			// It may be a bit too large in case of removals, but that's fine for now
-			Set<RowKey> keys = new HashSet<RowKey>( cleared ? currentState.size() : snapshot.size() + currentState.size() );
+			Set<RowKey> keys = new LinkedHashSet<RowKey>( cleared ? currentState.size() : snapshot.size() + currentState.size() );
```

Comment:
Same here. Let's create a `newLinkedHashSet` method in `CollectionHelper` and I'll optimize it.